#159328012 Create POST /auth/login endpoint

olusoladavid opened the pull request:
#### What does this PR do?

Adds an API endpoint to the app to help the user login to their account and then receive an authentication token as response

#### Description of Task to be completed

- Create SQL queries
- Create POST endpoint
- Validate user inputs
- Check if user exists
- Return JWT

#### How should this be manually tested?

- Pull branch into your local repo
- Run `npm run dev` in your local machine

Using Postman: 
Send a POST request to `/auth/login` with JSON body using the sample below:
```
{
"email": "validemail@gmail.com",
"password": "apassword"
}
```
#### What are the relevant pivotal tracker stories?

[#159328012](https://www.pivotaltracker.com/story/show/159328012)

olusoladavid committed:
feat(Auth): create POST /auth/signup endpoint

- Create SQL queries
- Create POST endpoint
- Validate user inputs
- Check for possible existing user
- Return JWT

[Delivers #159327500]

olusoladavid committed:
feat(Auth): Create POST /auth/login endpoint

- Create SQL queries
- Create POST endpoint
- Validate user inputs
- Return JWT

[Delivers #159328012]

FredDominant commented on server/routes/index.js:
```diff
@@ -13,46 +9,31 @@ router.get('/', (req, res) => {
-    check('content').isString(),
-    check('isFavorite').isBoolean(),
-    check('isPublic').isBoolean(),
+    check('email')
```
why don't you have these validations as a function/middleware so it becomes easier to read?

olusoladavid replied:
Okay. Thanks for the feedback. I will refactor accordingly.

olusoladavid committed:
chore(Validation): refactor validator middleware into utils folder

- create validate.js file
- move all middleware validation code into validate.js

olusoladavid committed:
Merge branch 'ft-signup-endpoint-159327500' into ft-login-endpoint-159328012

olusoladavid committed:
chore(Validation): refactor validation middleware into utils folder

- create validate.js in utils folder
- move all validation code into validate.js
